packages/auth/amplify_auth_cognito/example/lib/main.dart

```diff
@@ -1,337 +1,143 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:amplify_api/amplify_api.dart';
 import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_authenticator/amplify_authenticator.dart';
 import 'package:amplify_flutter/amplify_flutter.dart';
-import 'Widgets/ConfirmResetWidget.dart';
-import 'Widgets/ConfirmSignInWidget.dart';
-import 'Widgets/ConfirmSignUpWidget.dart';
-import 'Widgets/SignInWidget.dart';
-import 'Widgets/SignUpWidget.dart';
-import 'Widgets/UpdatePasswordWidget.dart';
-import 'Widgets/ViewUserAttributes.dart';
+import 'package:flutter/material.dart';
+
 import 'amplifyconfiguration.dart';
 
 void main() {
-  runApp(MaterialApp(home: MyApp()));
+  runApp(const MyApp());
 }
 
 class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
   @override
-  _MyAppState createState() => _MyAppState();
+  State<MyApp> createState() => _MyAppState();
 }
 
 class _MyAppState extends State<MyApp> {
-  final usernameController = TextEditingController();
-  final passwordController = TextEditingController();
-  final emailController = TextEditingController();
-  final phoneController = TextEditingController();
-  final confirmationCodeController = TextEditingController();
-  final oldPasswordController = TextEditingController();
-  final newPasswordController = TextEditingController();
-  late StreamSubscription subscription;
-
-  bool _isAmplifyConfigured = false;
-  late AmplifyAuthCognito auth;
-  String displayState = '';
-  String authState = 'User not signed in';
-  String lastHubEvent = '';
-  AmplifyException? _error;
-
   @override
   void initState() {
     super.initState();
+    _configure();
   }
 
-  void showResult(_authState) async {
-    setState(() {
-      _error = null;
-      authState = _authState;
-    });
-    print(authState);
-  }
-
-  void changeDisplay(_displayState) async {
-    setState(() {
-      _error = null;
-      displayState = _displayState;
-    });
-    print(displayState);
+  Future<void> _configure() async {
+    try {
+      await Amplify.addPlugin(AmplifyAPI());
+      await Amplify.addPlugin(AmplifyAuthCognito());
+      await Amplify.configure(amplifyconfig);
+      safePrint('Successfully configured Amplify');
+    } on Exception catch (e) {
+      safePrint('Configuring Amplify failed: $e');
+    }
   }
 
-  void setError(AmplifyException e) async {
-    setState(() {
-      _error = e;
-    });
+  @override
+  Widget build(BuildContext context) {
+    return Authenticator(
```

Comment:
🔥

Reply:
I think if we are going to add the Authenticator here, we should implement the non-sign/sign-up APIs within `MyHomePage` (i.e. fetchAuthSession, getCurrentUser, changePassword, device/attribute management, etc).

Also why are we keeping the widgets being replaced by the authenticator? Can't we just delete them? (Maybe I'm missing something here.)

Reply:
Those screens will be added back as we add the API implementations. None of those APIs have been moved over yet, though.